### PR TITLE
Added renderOrder to convert functions

### PR DESCRIPTION
**Pull request by ahcorde**
This PR is part of the effort to avoid z-fighting between coplanar meshes. 

Related with https://github.com/ignitionrobotics/ign-rendering/pull/188

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: src/Conversions.cc [hunk 1]
```diff
@@ -262,6 +262,7 @@ msgs::Material ignition::gazebo::convert(const sdf::Material &_in)
   msgs::Set(out.mutable_diffuse(), _in.Diffuse());
   msgs::Set(out.mutable_specular(), _in.Specular());
   msgs::Set(out.mutable_emissive(), _in.Emissive());
+  out.set_render_order(_in.RenderOrder());
```
Comment (chapulina) [inline src/Conversions.cc:265]: This needs to target `main`, because the `ign-rendering` PR breaks ABI.

**Review by chapulina — CHANGES_REQUESTED** [review]

**Comment by ahcorde** [comment]
Retargeted to `main`

**Comment by chapulina** [comment]
This is blocked by upgrading `ign-gazebo5` to use `ign-msgs7`: https://github.com/ignition-tooling/release-tools/issues/362

**Review by chapulina — COMMENTED** [review]
Works for me. Mind setting and checking the render order on some tests in `Conversions_TEST`?

**Review by chapulina — APPROVED** [review]
Approved, but it can't be merged until the upgrade to `ign-msgs7` is complete.

**Review by iche033 — APPROVED** [review]

**Comment by chapulina** [comment]
The `ign-msgs7` upgrade is in, but now we need an SDF10 release: https://github.com/osrf/sdformat/pull/461

File: src/Conversions_TEST.cc [hunk 2]
```diff
@@ -237,6 +238,7 @@ TEST(Conversions, Material)
   EXPECT_EQ(math::Color(1.3f, 1.4f, 1.5f, 1.6f),
       msgs::Convert(materialMsg.emissive()));
   EXPECT_TRUE(materialMsg.lighting());
+  EXPECT_DOUBLE_EQ(2.5, pbrMsg.render_order());
```
Comment (chapulina) [inline src/Conversions_TEST.cc:241]: This is failing to compile

```suggestion
  EXPECT_DOUBLE_EQ(2.5, materialMsg.render_order());
```

**Review by chapulina — CHANGES_REQUESTED** [review]

**Review by chapulina — APPROVED** [review]
LGTM, just needs the homebrew bottle to be finished to fix CI: https://github.com/osrf/homebrew-simulation/pull/1307

**Comment by ahcorde** [comment]
@osrf-jenkins retest this please

**Comment by ahcorde** [comment]
This failing test is unrelated ServerRepeat/SceneBroadcasterTest.State to this PR. Merging

**Pull request merged**